nodeadm/internal/system/networking.go

```diff
@@ -83,26 +95,86 @@ func (a *networkingAspect) ensureEKSNetworkConfiguration(cfg *api.NodeConfig) er
 	if err := os.WriteFile(eksPrimaryENIOnlyConfPathName, eksPrimaryENIOnlyConfContent, networkConfFilePerms); err != nil {
 		return fmt.Errorf("failed to write eks_primary_eni_only network configuration: %w", err)
 	}
-	if err := a.reloadNetworkConfigurations(); err != nil {
-		return fmt.Errorf("failed to reload network configurations: %w", err)
-	}
 	return nil
 }
 
-// eksPrimaryENIOnlyTemplateVars holds the variables for eksPrimaryENIOnlyConfTemplate
-type eksPrimaryENIOnlyTemplateVars struct {
+type primaryInterfaceTemplateVariables struct {
 	PermanentMACAddress string
 }
 
-// generateEKSPrimaryENIOnlyConfiguration generates the eks primary eni only network configuration.
-func (a *networkingAspect) generateEKSPrimaryENIOnlyConfiguration(cfg *api.NodeConfig) ([]byte, error) {
+func generatePrimaryInterfaceNetworkConfiguration(cfg *api.NodeConfig) ([]byte, error) {
 	primaryENIMac := cfg.Status.Instance.MAC
-	templateVars := eksPrimaryENIOnlyTemplateVars{
+	templateVars := primaryInterfaceTemplateVariables{
 		PermanentMACAddress: primaryENIMac,
 	}
 
 	var buf bytes.Buffer
-	if err := eksPrimaryENIOnlyConfTemplate.Execute(&buf, templateVars); err != nil {
+	if err := primaryInterfaceTemplate.Execute(&buf, templateVars); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+// ensureAdditionalInterfaceNetworkConfigurations configures the non-zero card interfaces in a way that mimics the
+// default AL2023 configuration. Non-zero card interfaces are not managed by vpc-cni and we're creating
+// systemd-networkd .network files for each interface.
+func (a *networkingAspect) ensureAdditionalInterfaceNetworkConfigurations(cfg *api.NodeConfig) error {
+	routeTableId := 10101
+	routeTableMetric := 613
+
+	for _, card := range cfg.Status.Instance.NetworkCards {
+		if card.Index == 0 {
+			continue
+		}
+
+		networkInterfaceConfName := fmt.Sprintf("70-%s.network", card.InterfaceId)
```

Comment:
Nit, since the primary one is using `80-ec2.network`(which overrides the same named one by `ec2-net-utils`
Maybe we should name it as `80-<eniID>.network`. so that it's the same priority 80.